pulsar-client-cpp/python/pulsar/schema/schema_avro.py

```diff
@@ -45,14 +46,23 @@ def _get_serialized_value(self, x):
         def encode(self, obj):
             self._validate_object_type(obj)
             buffer = io.BytesIO()
-            m = {k: self._get_serialized_value(v) for k, v in obj.__dict__.items()}
+            m = self.encode_dict(obj.__dict__)
             fastavro.schemaless_writer(buffer, self._schema, m)
             return buffer.getvalue()
 
+        def encode_dict(self, d: dict):
+            obj = {}
+            for k, v in d.items():
+                if isinstance(v, Record):
+                    obj[k] = self.encode_dict(v.__dict__)
```

Comment:
This check could also be don in the `_get_serialized_value()` method:

```python
def _get_serialized_value(self, x):
            if isinstance(x, enum.Enum):
                return x.name
            elif isinstance(x, Record):
                self.encode_dict(x.__dict__)
            else:
                return x
```

Reply:
I fixed this. PTAL